airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -1272,20 +1273,28 @@ def create_async_retriever(
             if model.abort_requester
             else None
         )
+        delete_requester = (
+            self._create_component_from_model(model=model.delete_requester, decoder=decoder, config=config, name=f"job delete - {name}")
+            if model.delete_requester
+            else None
+        )
         status_extractor = self._create_component_from_model(model=model.status_extractor, decoder=decoder, config=config, name=name)
         urls_extractor = self._create_component_from_model(model=model.urls_extractor, decoder=decoder, config=config, name=name)
         job_repository: AsyncJobRepository = AsyncHttpJobRepository(
             creation_requester=creation_requester,
             polling_requester=polling_requester,
             download_retriever=download_retriever,
             abort_requester=abort_requester,
+            delete_requester=delete_requester,
             status_extractor=status_extractor,
             status_mapping=self._create_async_job_status_mapping(model.status_mapping, config),
             urls_extractor=urls_extractor,
         )
 
         return AsyncRetriever(
-            job_orchestrator_factory=lambda stream_slices: AsyncJobOrchestrator(job_repository, stream_slices),
+            job_orchestrator_factory=lambda stream_slices: AsyncJobOrchestrator(
+                job_repository, stream_slices, JobTracker(5)
```

Comment:
how do we (or the connector developer) should decide on 5 vs 10?

Reply:
This would be the limit of concurrent jobs on the server side assuming the API token is not re-used for other processes that creates jobs. Right now, I blindly set this to 5 but we can make it configurable. This will be a bit different than the usual though as this is a limit across streams so probably at the root level of the manifest